File: GridSortMax/solve/cpp/GridSortMaxSolver.cpp
#include <fstream>
#include <topcoder>
#include "GridSortMax.cpp"
namespace tc = TopCoder;

int main(int argc, char const *argv[]) {
  try {
    std::ifstream ifs(argv[1]);
    int n; tc::read(ifs, n); tc::next(ifs);
    int m; tc::read(ifs, m); tc::next(ifs);
    vector<int> grid; tc::read(ifs, grid);
    ifs.close();

    std::ofstream ofs(argv[2]);
    GridSortMax solver;
    tc::show(ofs, solver.findMax(n, m, grid));
    ofs.close();
  } catch (std::exception &e) {
    std::cerr << e.what() << std::endl;
  }
  return 0;
}

File: GridSortMax/solve/ruby/GridSortMaxSolver.rb
#! /usr/bin/env ruby

require_relative "GridSortMax"

require "gettc/types"
include Gettc

def init
  gettc_home = File.expand_path(ENV["GETTC_HOME"] || File.join(ENV["HOME"], ".gettc"))
  $LOAD_PATH << File.join(gettc_home, "include", "ruby")
  require "topcoder"
  include TopCoder
end

def main
  reader = Reader.new(IO.read(ARGV[0]))
  n = reader.next(TInt)
  reader.next()
  m = reader.next(TInt)
  reader.next()
  grid = reader.next(TArray.new(TInt))

  result = GridSortMax.new().findMax(n, m, grid)
  IO.write(ARGV[1], Writer.new.next(result, TString).to_s)
end

init
main
